agent/client/client.go

```diff
@@ -51,8 +52,13 @@ const (
 	backoffMaxDelay      = 15 * time.Second
 	clockDriftWarning    = 5 * time.Second
 	defaultActionTimeout = 10 * time.Second // default timeout for compatibility with an older server
+
+	cacheSize                   = 30_000
+	defaultWindowConnectionTime = "24h"
 )
 
+var requestConnectedUpTimeCache gcache.Cache
```

Comment:
This impl doesn't scale.
I think we can do better, we can store only failed attempts. In sorted by time data structure.
And in over case, we need to store data sequentially, and it will be ordered.

And we don't need any cache here.

Reply:
Well, I added cache there only for reason to keep connection attempts data during set period of time (by default for 24 hours) which is used for calculation connected up time value. And this period of time is equalt to cache expiration time.

The thing there that intervals between connection to server are not equal, for example requests between dial requests are between 0 and 18.75 seconds.

Reply:
and using cache simplify of removing data about connection attempts from the memory which is not needed (which is not in the specified time window during which we count connected up time).

Reply:
@cezmunsta 

> if you only store failed attempts then how can you calculate a percentage?
we need to know when we started, we know that we have fixed interval to check connection (ie. 10sec), in this case we need to know only failed attempts to calculate uptime

Reply:
@ritbl is that true? The interval is random and so how do you know how many attempts were made?

Reply:
I think we have a heartbeat with constant intervals, that can be used. Or we can track when connection have been lost. In any case, I think it is better to track time when we have no connection.